src/wasm.rs

```diff
@@ -134,6 +134,39 @@ pub trait Wasm<ExecC, QueryC> {
 
     /// Returns a raw state dump of all key-values held by a contract with specified address.
     fn dump_wasm_raw(&self, storage: &dyn Storage, address: &Addr) -> Vec<Record>;
+
+    /// Returns the namespace of the contract storage.
+    fn contract_namespace(&self, contract: &Addr) -> Vec<u8> {
+        let mut name = b"contract_data/".to_vec();
+        name.extend_from_slice(contract.as_bytes());
+        name
+    }
+
+    /// Returns **read-only** (not mutable) contract storage.
+    fn contract_storage<'a>(
+        &self,
+        storage: &'a dyn Storage,
+        address: &Addr,
+    ) -> Box<dyn Storage + 'a> {
```

Comment:
I'd think of a new trait in MT that would avoid exposing mutable methods of cosmwasm's `Storage`. Something like:
```rust
pub trait ReadOnlyStorage: Storage { ... }
```

Reply:
Yes, I thought about it, but anything that should be accessible via `App` must unfortunately go to public API of `Wasm`, unless we block exchanging `Wasm` in `App`. But the idea is ok.